charts/datadog/templates/_container-agent.yaml

```diff
@@ -92,7 +92,7 @@
     {{- include "containers-cluster-agent-env" . | nindent 4 }}
     {{- end }}
     - name: DD_APM_ENABLED
-      value: "false"
+      value: {{  (default false (or .Values.datadog.apm.enabled .Values.datadog.apmEnabled)) | quote }}
```

Comment:
the DD_APM_ENABLED has no incident on the `agent` container. it should only impact the `trace-agent` container.

[Here](https://github.com/DataDog/helm-charts/blob/main/charts/datadog/templates/_container-trace-agent.yaml#L38) you can see that the value is hardcoded to `true`. 

The current logic for enabling APM with the chart is:
* APM is enable by default with UDS (Unix Domain Socket) communication.
* To have the possibility to send APM trace with a TCP communication, the option `datadog.apm.portEnabled` need to be set to `true`.
   see: https://github.com/DataDog/helm-charts/tree/main/charts/datadog#enabling-apm-and-tracing

Reply:
@clamoriniere I have datadog.apm.portEnabled set to true which sets up DD_APM_ENABLED environment variable to true for trace-agent and on agent containers it's set to false. In such configuration the application throws an error `Cause: Errno:ECONNREFUSED Failed to open TCP connection to 10.0.54.163:8126`. If i will manually edit DD_APM_ENABLED env variable on the replica set and configure agent container with DD_APM_ENABLED = true value everything is working as expected. What else can cause this issue?